### ISSUE TITLE
No ID should be invalid

### ISSUE THREAD

**Issue by peterbe** [mdn/kuma#5987]
If you paste this in "source mode" in the WYWIWYG editor and press "Publish"
```html
<div id="123">123</div>

<div id="brac(ket)">brac(ket)</div>

<div id="fullstop.">fullstop.</div>
```
The output is exactly the same. 
But these are invalid DOM IDs. At least if you use...
```javascript
document.querySelector(`#${id}`)
```
...as the measure. 

What should happen is that somewhere in Python it should clean that up and the output should be something like:

```html
<div id="id123">123</div>

<div id="brac_ket_">brac(ket)</div>

<div id="fullstop">fullstop.</div>
```

Note! We probably already have some solutions that do some of those cleaning tasks if the `tagName` is an `H2` or `H3`. Perhaps the initial solution is simply to add `DIV` to that lists of tag names. 

Also, note that there might be KumaScripts that rely on the IDs as the author entered them. E.g. `{{SomeMacro("brac(ket)")}}` and `SomeMacro.ejs` might expect to be able to do something like `<a name="brac(ket)">`.

**Comment by tobinmori** [comment]
Any idea how un/often this occurs @peterbe?

**Comment by peterbe** [comment]
It's pretty bad! I download all of MDN to disk using the Document JSON API. 
Then I wrote a Python script to analyze every JSON doc. Here's the output when I run it for `mdn-api-prod/api/v1/doc/en-US/Web/`

```
Analyzed 8,487 files.
1,536 NOT valid IDs, 78,554 valid.
1.9% are not valid
```

And for `mdn-api-prod/api/v1/doc/en-US/Web/` it's 
```
Analyzed 3,836 files.
2,093 NOT valid IDs, 41,618 valid.
4.8% are not valid
```

The script also prints all offending IDs (once per ID) and that gives you a preview of them:

`en-US` https://gist.github.com/peterbe/fc86ece0cd19a8f0a82b4f46112836ad
`fr` https://gist.github.com/peterbe/0657f3965ed3f9b22bc1b26b36953aa1

[Here's the test I used](https://gist.github.com/peterbe/f8acd5233efa5f18732cd0a0a9a648ba#file-valid_ids-py-L16-L28)

I [copied some of these to Codepen](https://codepen.io/peterbe/pen/xxxgrqw?editors=1011) to sanity check that it does throw an error. 

However, not all of these pages actually cause any problems. It's only on pages that have live samples. 
I ran out of time but tomorrow I'll write a thing that, for every `bodyHTML` that probably have live samples, print their URL and count them as that would indicate how many pages are causing error boundaries because of invalid IDs.

**Comment by peterbe** [comment]
One curious thing, that I just learned is that these exceptions are only thrown on `document.querySelector[All]('#BADID')`.
E.g. 
```
>> document.getElementById('123(blur)')
null
>> document.querySelector('#123(blur)')
SyntaxError: '#123(blur)' is not a valid selector
```

Also, I dug and dug and dug and tried to reverse backwards from how `live-examples.js` works to try to find pages that when opened would trigger the error boundaries. I failed. Couldn't find a single page. 

Also, my initial Python script isn't right. It would complain about `Un_dégradé_à_40°` being invalid but it's actually not a problem. 
```
>> document.querySelector('#Un_dégradé_à_40°')
{a thing}
```

All in all, I think there are tonnes of invalid IDs all over the place but I can't find a single one where the `document.querySelector()` calls done is `live-examples.js` trigger.
